### PR TITLE
Added support to avoid quotation character

### PR DESCRIPTION
**Pull request by pmaiolino**
When you need to export a csv file, it is possible that it is to add quotation characters only for some field. 
In DelimitedRecordReader class it was added the property UseQuoteCharacter to manager these cases.

If this request will be accepted, will it be possible to add it to the last NuGet package?

### CODE REVIEW

File: src/NLight/IO/Text/DelimitedRecordWriter.cs [hunk 1]
```diff
@@ -14,12 +14,14 @@ public class DelimitedRecordWriter
...
 
+		public bool UseQuoteCharacter { get; set; } = true;
+		
 		private char _quote;
 		private char _quoteEscape;
 
 		private string _escapedQuoteString;
 		private string _quoteString;
-
+
```
Comment (slorion) [inline src/NLight/IO/Text/DelimitedRecordWriter.cs]: Remove whitespaces

File: src/NLight/IO/Text/DelimitedRecordWriter.cs [hunk 2]
```diff
@@ -81,13 +83,19 @@ protected override void WriteFieldCore(TextWriter writer, string value)
 				value = value.TrimEnd();
 			else if ((this.FieldTrimmingOptions & TrimmingOptions.Start) == TrimmingOptions.Start)
 				value = value.TrimStart();
+			
+			if (UseQuoteCharacter)
+            {
```
Comment (slorion) [inline src/NLight/IO/Text/DelimitedRecordWriter.cs]: Fix indentation (with tabs)

File: src/NLight/IO/Text/DelimitedRecordWriter.cs [hunk 3]
```diff
@@ -81,13 +83,19 @@ protected override void WriteFieldCore(TextWriter writer, string value)
...
+				value = value.Replace(_quoteString, _escapedQuoteString);
+				
+				writer.Write(_quote);
+				writer.Write(value);
+				writer.Write(_quote);
+			}
+			else
```
Comment (slorion) [inline src/NLight/IO/Text/DelimitedRecordWriter.cs:95]: Add curly braces

**Comment by slorion** [comment]
Thank you for your contribution! Although always quoting fields should not be a problem and it makes the output coherent, the writer can give the option of opting out. What is your use case for that feature?

**Comment by pmaiolino** [comment]
Thank you for your consideration!
In my case I neeed to export csv file respect to a standard in which some fields are string and other are number.
Who has defined this standard decided that the quotation character is needed only for strings, for numbers not.
Generally a char can not be empty, so unfortunally without my changes I didn't found a way to follow this standard.

**Comment by slorion** [comment]
Would it not be better in that case to specify the property `UseQuoteCharacter` per column? I did not envision the writer having to check column metadata when writing field values, but that said, the writer could keep track of what is the current column being written and look up metadata that way, i.e. each time a field is written, current column index is incremented, and when writing the end of a record, the index is reset.

**Comment by pmaiolino** [comment]
This could be fine if all records follow the same pattern but, for example in the standard I was writing about, there are 3 different schemes. You can see an example in this file [Example.txt](https://github.com/slorion/nlight/files/1811284/Example.txt).
The first record is a header and it contains metadata, the second contains the columns' title (in particular strings without quotes) and finally other records follow a third scheme.
So, I thought that with a single property there is more flexibility for the programmer to define even complex schemas like in this case.

**Comment by slorion** [comment]
Well, your example is not a valid CSV file since it is supposed to have the same number of fields per record, but I get your point. I will sleep on it and come back to you. Feel free to add any other insights that could be useful.

**Comment by slorion** [comment]
Ok, having a global property is fine. Should the need arise to have a per column setting, that global property could then act as a default value. This pattern is already used in NLight.

So, could you make the changes related to my comments on your commit? Then I will merge and publish a new version. Thank you!

**Comment by pmaiolino** [comment]
Well, I'm glad for this! I pushed the changes about your review comments but, I did not understand if you would like me to add other changes.

**Comment by slorion** [comment]
No other changes, I was just thinking out loud about how your PR could be extended in the future if a per column property should be needed. Thank you again, will make a new release asap.

**Comment by pmaiolino** [comment]
Thank you so much!

**Pull request merged**